src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -60,14 +68,19 @@ public abstract partial class AbstractDocumentQuery<T, TSelf> : IDocumentQueryCu
         public string CollectionName { get; }
 
         private int _currentClauseDepth;
-
+        
         protected string QueryRaw;
-
+        
+        internal bool IsFilterActive { get { return ModeStack.Any() && ModeStack.Peek() is true; } }
+        
+        protected Stack<bool> ModeStack = new ();
```

Comment:
`ModeStack` -> `FilterModeStack` so someone who reads the code will immediately know it's about the filter operation

Reply:
done